builtin/providers/aws/resource_aws_eip_association_test.go

```diff
@@ -0,0 +1,118 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSEIPAssociation_basic(t *testing.T) {
+	var i ec2.Instance
+	var a ec2.Address
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSEIPAssociationDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccAWSEIPAssociationConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckInstanceExists(
+						"aws_instance.foo.0", &i),
+					testAccCheckAWSEIPExists(
+						"aws_eip.bar.0", &a),
+					testAccCheckAWSEIPAssociationExists(
+						"aws_eip_association.by_allocation_id", &i, &a),
+					testAccCheckInstanceExists(
+						"aws_instance.foo.1", &i),
+					testAccCheckAWSEIPExists(
+						"aws_eip.bar.1", &a),
+					testAccCheckAWSEIPAssociationExists(
+						"aws_eip_association.by_public_ip", &i, &a),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSEIPAssociationExists(name string, i *ec2.Instance, a *ec2.Address) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		if rs.Primary.Attributes["allocation_id"] == *a.AllocationId {
+			if a.InstanceId != nil && rs.Primary.Attributes["instance_id"] == *a.InstanceId {
+				// success
+				return nil
+			}
+		}
+
+		return fmt.Errorf("Error finding instance/address")
+	}
+}
+
+func testAccCheckAWSEIPAssociationDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		log.Printf("\n\n----- This is never called")
```

Comment:
Is this debugging, or comments? It should be removed for the former, a comment for the latter ... 

Shouldn't we check that the association is gone? E.g. query the API and make sure this EIP no longer has an association? Isn't that "destroyed", in this context?